knowledge-base/grid-hide-vertical-scrollbar.md

```diff
@@ -0,0 +1,86 @@
+---
+title: How to Remove the Vertical Grid Scrollbar
+description: How to hide the vertical scrollbar of the Grid and TreeList with CSS, if scrolling is not used.
+type: how-to
+page_title: How to Hide the Vertical Grid Scrollbar
+slug: grid-hide-vertical-scrollbar
+position: 
+tags: grid, treelist, scroll, scrollbar, scrolling
+ticketid: 1541813, 1541728
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>Grid for Blazor, TreeList for Blazor</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+
+How do I hide the Grid vertical scrollbar? I don't set the Grid `Height`, so I prefer to completely remove the disabled vertical scrollbar with CSS.
+
+## Solution
+
+There are three required CSS styles to disable vertical scrolling and ensure proper cell alignment between the header, data and footer area of the Grid/TreeList:
+
+1. Remove the vertical scrollbar, which is visible by default.
+1. Remove the empty space (`padding`) above the vertical scrollbar, in the **header** area.
+1. Remove the same space below the vertical scrollbar, in the **footer** area.
+
+The exact same approach can be used for the **TreeList** as well, as the TreeList reuses Grid CSS classes.
+
+>caption Hide the Grid/TreeList vertical scrollbar
+
+````CSHTML
+@* Hide the vertical scrollbar of the Grid *@
+
+<TelerikGrid Data="@GridData" Class="grid-no-scroll">
+    <GridColumns>
+        <GridColumn Field=@nameof(Product.Name) Title="Product Name">
+            <FooterTemplate>
+                footer
+            </FooterTemplate>
+        </GridColumn>
+    </GridColumns>
+</TelerikGrid>
+
+<style>
+    .grid-no-scroll .k-grid-header,
+    .grid-no-scroll .k-grid-footer {
+        padding-right: 0;
+    }
+    .grid-no-scroll .k-grid-content {
+        overflow-y: auto;
```

Comment:
if set to `auto` and the `Height` of the grid does not accommodate all rows, you will still get a scrollbar in the content which will put the headers and footers out of sync with the data cells. Perhaps should be `hidden`, or he caveat explained.

Reply:
Indeed, an explanation is good. The idea here is that even if the developer messes up the Grid declaration by setting a Height, the user will still be able to see all rows. The other option is to set `overflow:visible`, but that will look uglier (if the problem occurs).